Kids news, kids books, and kids movies instant answers

dogomedia opened the pull request:
DuckDuckHack Pull Request

# Spice Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?**

We have added 3 instant answers:
1) Kids news - Uses media template to display kids news and current events from the DOGOnews API.
2) Kids books - Uses media template to display kids books and ratings from the DOGObooks API.
3) Kids movies - Uses media template to display kids movies and ratings from the DOGOmovies API.

**What problem does your Instant Answer solve (Why is it better than organic links)?**

The instant answers wer provide shows only results that are relevant to kids.
1) For kids news, the DOGOnews news articles are written specifically with kids in mind, using vocabulary and language appropriate for kids.
2) For kids books, we only return books that are appropriate for children and young adults.
3) For kids movies, we only return movies that are appropriate for children (G, PG, and PG-13).

**What is the data source for your Instant Answer? (Provide a link if possible)**

We developed the API for these instant answers to search our own database (http://www.dogonews.com, http://www.dogobooks.com, http://www.dogomovies.com)

**Why did you choose this data source?**

We own the www.dogonews.com, www.dogobooks.com, and www.dogomovies.com websites.

**Are there any other alternative (better) data sources?**

Since we are one of the leading destinations for kids news and media, we believe that we are the best source of data.

**What are some example queries that trigger this Instant Answer?**

To get the latest news search for 'kids news'.  To get for articles related to 'obama' or 'nasa', you can search 'kids articles on nasa'.  To get for science articles, enter 'kids science articles'.

To get the most popular kids books, search for 'kids books'. To get books related to 'Harry Potter' search 'harry potter kids books'.

To get the most popular kids movies, search for 'kids movies'. To get books related to 'Hunger Games' search 'hunger games kids movie'.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

DOGOnews IA:
This will be especially useful for kids looking for current events and news articles to use for homework.  This is also useful for teachers that are looking for current event content to use in their classroom.

DOGObooks IA:
This is useful for kids that want to find out more about other kids books, or for parents looking for kid friendly books.

DOGOmovies IA:
This is useful for kids that want to find out more about other kids movies, or for parents looking for kid friendly movies.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

No

**Which existing Instant Answers will this one supersede/overlap with?**

1. DOGOnews IA may overlap with the News IA since the triggers both contain "news".
2. DOGObooks IA may overlap with the Book IA since the triggers both contain "book review".
2. DOGOmovies IA may overlap with the Movie IA or the About IA since the trigger "the hunger games movie" will show our results.

**Are you having any problems? Do you need our help with anything?**

For now I think we are satisified with the look of the instant answer, but in the future we may want to further customize the templates.

We would also like to get an idea of how much traffic we are likely to see hitting our API and how to best set the cache settings.

**What are the terms of use for the API? Will DuckDuckGo need specific authorization (e.g. an API key)? Are there any costs associated with API usage?**

An API key is required, which we will provide to DuckDuckGo with no cost.

**Where did you hear about DuckDuckHack? (For first time contributors)**

Google

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

News - http://i.imgur.com/9oVaMa3.jpg
Books - http://i.imgur.com/1w4sQFk.jpg
Movies - http://i.imgur.com/bXeEYY6.jpg

## Checklist
Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information
[X] Wrote test file and added to t/ directory
[X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [X] Google Chrome
        [X] Firefox
        [X] Opera
        [X] IE 10

    - Windows 7
        [X] Google Chrome
        [X] Firefox
        [X] Opera
        [X] IE 8
        [X] IE 9
        [X] IE 10

    - Windows XP
        [] IE 7
        [X] IE 8

    - Mac OSX
        [X] Google Chrome
        [X] Firefox
        [X] Opera
        [X] Safari

    - iOS (iPhone)
        [X] Safari Mobile
        [X] Google Chrome
        [] Opera

    - iOS (iPad)
        [X] Safari Mobile
        [X] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [X] Native Browser
        [X] Google Chrome
        [] Opera
```

----
> IA Pages: https://duck.co/ia/view/dogo_books, https://duck.co/ia/view/dogo_movies, https://duck.co/ia/view/dogo_news

moollaza commented:
Hey @dogomedia!

Thanks for the pull request! Now's when the fun begins!

We work closely with every contributor to make Instant Answers the best they can be so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and
community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask our staff or community:

### Community Leaders:

- MrChrisW
- mattr555
- mintsoft
- killerfish
- bradcater
- javathunderman
- kablamo
- loganom
- TomBebbington

### Staff:

- @moollaza (me)
- jagtalon
- abeyang
- chrismorast
- russellholt
- (and a few others)[https://github.com/orgs/duckduckgo/teams/owners]

------

Just in case, you can also reach us on [IRC](http://webchat.freenode.net/?channels=duckduckgo) or ask the community through our DuckDuckHack [Developer Email list](https://www.listbox.com/subscribe/?list_id=197814).

>More Info: https://duck.co/duckduckhack/review_process

moollaza commented:
@dogomedia apologies for the delay! I'll be looking at this today. Thanks again for submitting this. We truly appreciate your contributions :smiley:

moollaza commented:
@dogomedia this is a great start! I'll be making comments in line, but for now my general feedback is:

 - we should reign in the triggers to be more specific, so they contain either "dogo" or "kids" to ensure the user intent is clear. The triggers we previously discussed were "dogo books", "kids books", "dogo movies", "kids movies", "dogo news and "kids news".  After the IA has launched we can revisit the triggers and broaden them if necessary.

- remove the custom design work you've implemented. We appreciate the time and effort you spent customizing the design, but all our Instant Answers should look and feel the same to ensure the user experience is consistent. Our design team has spent time designing the built-in templates to ensure all our Instant Answers look and function great across all major browsers and devices.

I'll go through the code and point out any more specific feedback but for now let's simply things and see how the IA looks and functions. Once the code is signed off the design team will give any feedback on suggested changes, if necessary.

Thanks!

moollaza commented on lib/DDG/Spice/DogoBooks.pm:
```diff
@@ -0,0 +1,34 @@
+
+triggers any => "dogobooks", "dogo books", "book", "books";
+
+spice to => 'http://api.dogomedia.com/api/v2/books/search.json?query=$1&api_key={{ENV{DDG_SPICE_DOGO_BOOKS_APIKEY}}}';
```
Can we normalize the API key to `DDG_SPICE_DOGO_APIKEY`? That would simplify things a bit on our end.

dogomedia commented:
@moollaza Thanks for the feedback.  

1. We are using regex triggers with a regex guard to avoid having to list all the various combinations of these terms.  We've made the triggers more specific now so that our IA's will initially trigger only on "dogo" and "kids", but we aslo added a few variations for the initial trigger:

a. For the news IA's initial trigger we added "children", "child", and "student" in addition to "kids".

b. For the book and movie IA's intial triggers, we added "children" and "child" in addition to "kids".

c. For the news IA's regex guard, we added "newspaper" and "current events" in addition to "news".  "Current events" may require some explanation - we've found that the majority of people who are looking for news content for kids uses "current events" and "news" interchangeably.

d. For the movies IA regex guard, we added "trailer", "dvd", and "film" in addition to "movie".

2. We have removed the custom design work. The news styling is now the standard media template styles. The movie and book templates now use the media styling for the existing "In Theaters" or "Movies" Instant Answers.

dogomedia commented:
@moollaza We've now normalized the API key to ```DDG_SPICE_DOGO_APIKEY```

moollaza commented on share/spice/dogo_books/dogo_books.js:
```diff
@@ -0,0 +1,48 @@
+                    ratingText: item.comments_count + ' reviews',
+                    reviewCount: item.comments_count,
+                    url: item.url,
+                    abstract: Handlebars.helpers.ellipsis(item.summary, 200)
```
The ellipsis can be removed, we automatically apply an ellipsis with a limit of 400 chars

moollaza commented on share/spice/dogo_books/subtitle_content.handlebars:
```diff
@@ -0,0 +1,4 @@
+<img src="//cdn.dogomedia.com/assets/dogobooks-a85d1c60e272132b90829f032b4f20d0.png" alt="DOGObooks"/>
```
Let's simplify this. Please remove the custom Dogo logo and MPAA image. We can stick to plain text here and mimic the current Movies IA display:

> Runtime: 1 hr. 30 min. | PG

moollaza commented on share/spice/dogo_movies/dogo_movies.css:
```diff
@@ -0,0 +1,167 @@
+    background-color: #f89406;
+}
+
+.zci--dogo_movies .mpaa-rating {
```
This MPAA CSS can be removed.

moollaza commented on share/spice/dogo_movies/dogo_movies.css:
```diff
@@ -0,0 +1,167 @@
+}
+
+.zci--dogo_movies .btn.btn--primary {
+    background:#f7a000;
```
Please remove this, we'll stick to the default button styling.

moollaza commented on share/spice/dogo_movies/dogo_movies.js:
```diff
@@ -0,0 +1,60 @@
+                itemType: 'kids movies'
+            },
+            normalize: function(item) {
+                var thumb = item.thumb;
```
All image URL's get passed to our internal proxy function so the requests are made to DDG instead of a 3rd party, in order to protect our user's privacy. 

It looks like these image urls are breaking our (admittedly naive) image URL check, for now please append "http:" to the url so it get's proxied correctly:

```
var thumb = "http:" + item.thumb
```

dogomedia replied:
No problem.  We can append the ```"http:"``` on our backend, and keep the client code clean.

moollaza replied:
Perfect!

moollaza commented on share/spice/dogo_movies/dogo_movies.js:
```diff
@@ -0,0 +1,60 @@
+                    ratingText: item.comments_count + ' reviews',
+                    reviewCount: item.comments_count,
+                    url: item.url,
+                    abstract: Handlebars.helpers.ellipsis(item.summary, 200)
```
Same here, we already apply an ellipsis

moollaza commented on share/spice/dogo_movies/dogo_movies.js:
```diff
@@ -0,0 +1,60 @@
+    
+    // Convert minutes to hr. min. format.
+    // e.g. {{dogo_movies_convert_minutes_to_hr_min 90}} will return 1 hr. 30 min.
+    Handlebars.registerHelper("dogo_movies_convert_minutes_to_hr_min", function(runtime) {
```
If you'd like, this can be made a private JS function, and runtime can be set in the `normalize` function:

```js
env.ddg_spice_dogo_movies = function(api_result) {
    ...
    Spice.add({
        ....,
        normalize: function(item) {
            runtime: hrsToMin(item.runtime),
            ...
        },
        ...
    }
}

function hrsToMin (time) {
    ...
} ;
```

moollaza commented on share/spice/dogo_movies/subtitle_content.handlebars:
```diff
@@ -0,0 +1,16 @@
+<img src="//cdn.dogomedia.com/assets/dogomovies-6959aaba3ab072822498c8e557a0c19a.png" alt="DOGOmovies"/>
```
Let's simplify this. Please remove the custom Dogo logo and MPAA image/CSS. We can stick to plain text here and mimic the current Movies IA display:

> Runtime: 1 hr. 30 min. | PG

moollaza commented on share/spice/dogo_news/dogo_news.js:
```diff
@@ -0,0 +1,50 @@
+                options: {
+                    variant: 'poster',
+                    buy: Spice.dogo_news.buy,
+                    subtitle_content: Spice.dogo_news.subtitle_content,
```
If you add `rating: true` you'll see the stars display below the image in the tile view:

![2015-04-01_17h34_53](https://cloud.githubusercontent.com/assets/873785/6953215/e34d8d2e-d895-11e4-8d00-ea6725f931c6.png)

moollaza commented on share/spice/dogo_news/subtitle_content.handlebars:
```diff
@@ -0,0 +1,5 @@
+By {{author}}
```
Again, let's simplify this and remove the image assets:

```html
By {{author}}
<span class="detail__separator"> | </span>
<span>{{comments_count}} comments</span>
```

moollaza commented on share/spice/dogo_news/dogo_news.js:
```diff
@@ -0,0 +1,50 @@
+            source = $(script).attr("src"),
+            query = decodeURIComponent(source.match(/dogo_news\/([^\/]+)/)[1]);
+    
+        Spice.add({
```
I just remembered we have a built-in News template that we use for our own News Instant Answer. If we update the code here as follows it will use the correct template. You'll notice the images are gone. For now we'll leave them out but we're working to bring image support to the new template and once that happens we'll be able to use your images as well:

```js
DDG.require('moment.js', function(){
            Spice.add({
                id: 'dogo_news',
                name: 'Kids News',
                data: api_result.results,
                meta: {
                    sourceName: 'DOGOnews',
                    sourceUrl: 'http://www.dogonews.com/search?query=' + encodeURIComponent(query) + '&ref=ddg',
                    itemType: 'kids news articles'
                },
                normalize: function(item) {
                    var thumb = item.hi_res_thumb || item.thumb;
                    return {
                        title: item.name,
                        source: item.author,
                        url: item.url,
                        excerpt: item.summary,
                        relative_time: moment(item.published_at).fromNow()
                    };
                },
                templates:{
                    item: 'news_item'
                }
            });
        });
```

The result should look like this:

![2015-04-01_17h56_56](https://cloud.githubusercontent.com/assets/873785/6953627/8a675e94-d898-11e4-8f80-f85d0a57857d.png)

dogomedia replied:
With your code above, I'm getting an error with ```moment```:

```Uncaught ReferenceError: moment is not defined```

What do I need to do to get access to the moment helper?

moollaza replied:
@dogomedia are you sure you wrapped everything in the `DDG.require` at the top? That should be all that's required to import Moment.js

dogomedia replied:
@moollaza Thanks for that hint, yes we missed the require.

moollaza commented on share/spice/dogo_movies/dogo_movies.js:
```diff
@@ -0,0 +1,60 @@
+                options: {
+                    variant: 'poster',
+                    buy: Spice.dogo_movies.buy,
+                    subtitle_content: Spice.dogo_movies.subtitle_content,
```
If you add `rating: true` you'll see the stars display below the image in the tile view:

![2015-04-01_17h38_44](https://cloud.githubusercontent.com/assets/873785/6953233/fd68a194-d895-11e4-8f2d-2358c2123250.png)

moollaza commented on share/spice/dogo_news/dogo_news.css:
```diff
@@ -0,0 +1,132 @@
+.zci--dogo_news .tile--dogo_news  {
```
With the news template, this CSS file can be removed :)

moollaza commented:
>  We are using regex triggers with a regex guard to avoid having to list all the various combinations of these terms. We've made the triggers more specific now so that our IA's will initially trigger only on "dogo" and "kids", but we aslo added a few variations for the initial trigger:

@dogomedia regarding the triggers, thanks a lot for updating them, but please continue to use a `words` trigger instead of a Regex. Regexs are intended to used when words won't work (e.g. matching a 5 digit zipcode) because they are more taxing on the backend as we need to run all regex triggers against each query. The word triggers are much more efficient. 

You could use a `map` and array to build the triggers or even `slurp` a text file from the /share/ dir into an array if that makes things easier.

BTW, the variations (kids/children, current events/news) are also fine with me :+1: 

As well I think the generic queries, "kids books", "kids movies" and maybe some variations like "popular kids movies" should still make an API call for "popular" -- that feature is great!

dogomedia commented:
For the generic queries, we'll move the logic for handling "popular" and "latest" to the backend api server as well in order to keep the client code clean.

moollaza commented:
>For the generic queries, we'll move the logic for handling "popular" and "latest" to the backend api server as well in order to keep the client code clean.

That sounds great, thanks!

dogomedia commented:
@moollaza Quick question... we noticed that the News IA has the trigger "news".  If our news IA has the trigger "kids", then which IA will win if someone types "kids news"?  In general how should we handle these collisions?

moollaza commented:
> @moollaza Quick question... we noticed that the News IA has the trigger "news". If our news IA has the trigger "kids", then which IA will win if someone types "kids news"? In general how should we handle these collisions?

We can updated the News IA to ignore the query if it matches something for DogoNews, or simply give the DogoNews IA a higher priority. In general though, whichever Spice calls `Spice.add()` first wins the race.

dogomedia commented:
@moollaza Ok, I think we've addressed all the issues that you brought up.  We went through and double-checked each custom css style (especially the ones that apply to mobile) to make sure that it is absolutely necessary in order to match the styling of the official DDG movies IA.

There's one last thing we'd like to clean up - the DDG movies IA grid view uses tiles with the class ```tile--c--n``` but with our IA, the tiles are using the class ```tile--c```.  So we copied all the media query css for ```tile--c--n``` into our stylesheets so that our tiles would match the DDG movies IA tiles.


```
/* Begin tile width override - We need to match .tile--c--n styling.  */
.zci--dogo_books .tileview--grid .tile--dogo_books {
    width: 15.65%;
    margin-left: 0.5%;
    margin-right: 0.5%;
}

@media only screen and (max-width: 60em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        width: 19%;
    }
}

@media only screen and (max-width: 44em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        width: 24%;
    }
}

@media only screen and (max-width: 37.4em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        width: 32.3%;
    }
}

@media only screen and (max-width: 29.5em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        width: 48%;
        margin-left: 1%;
        margin-right: 1%;
    }
}

@media only screen and (min-width: 68.4em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        margin-left: 0.25%;
        margin-right: 0.25%;
        width: 13.75%;
    }
}

@media only screen and (min-width: 76em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        margin-left: 0.25%;
        margin-right: 0.25%;
        width: 13.75%;
    }
}

@media only screen and (min-width: 83.6em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        width: 11.98%;
    }  
}

@media only screen and (min-width: 100em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        width: 11.98%;
    }
}

@media only screen and (min-width: 110em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        width: 9.49%;
    }
}

@media only screen and (min-width: 125em) {
    .zci--dogo_books .tileview--grid .tile--dogo_books {
        margin-left: 0.125%;
        margin-right: 0.125%;
        width: 8.825%;
    }
}
/* End tile width override - We need to match .tile--c--n styling.  */
```

Ideally we could just have our tiles use ```tile--c--n```, but I'm not sure how to do that.  Both our IA and the DDG movie IA seem to be using ```group: 'media'``` and ```variant: 'poster'```.

Is there something else that we need to do to get our tiles to use the same tile class as the DDG movies IA tiles?  If so, then we could get rid of those media query css lines above.

moollaza commented:
@dogomedia good catch! It's a very subtle bug, but you need to set the variant outside the `options` block:

```js
options: { ... },
variants: {
    tile: 'poster'
}
```

This sets the correct `tile--c--n` class.

FWIW, I spoke with the team and we're going to build a new template for IA's like these that show a large poster as the tile. That way we can remove all this IA-Specific CSS and our global styles will handle everything.

dogomedia commented:
@moollaza Thanks for the ```tile--c--n``` fix.  We've removed all those duplicated media query css lines.

Two more questions:
1. Do you see anything else that needs to be corrected before the pull can be merged?
2. It's a shame to lose the thumbnails on the news IA.  Just curious what the timing is on updating your existing DDG news IA to include thumbnail support?

moollaza commented:
>1. Do you see anything else that needs to be corrected before the pull can be merged?

I think these are just about done :) @abeyang these are ready for a design pass. I'll put them on a dev machine shortly.

>2. It's a shame to lose the thumbnails on the news IA. Just curious what the timing is on updating your existing DDG news IA to include thumbnail support?

Unfortunately I don't know for sure, but our design team is eager to get the images showing, so it's definitely on our radar.

moollaza commented:
It's up on DDH8 for testing!

- https://ddh8.duckduckgo.com/?q=dogo+movies&ia=kidsmovies&iai=2
- https://ddh8.duckduckgo.com/?q=dogo+news+&ia=kidsnews
- https://ddh8.duckduckgo.com/?q=dogo+books&ia=kidsbooks

- https://ddh8.duckduckgo.com/?q=harry+potter+kids+book+review&ia=kidsbooks&iai=3 
- https://ddh8.duckduckgo.com/?q=lion+king+kids+movie+reviews&ia=kidsmovies&iai=1
- https://ddh8.duckduckgo.com/?q=entertainment+kids+news&ia=kidsnews

abeyang commented:
@dogomedia re: the book and movie IAs, as they currently stand, the tiles are a tad too long. I think we can get rid of the titles (as the poster/book covers reveal them), and let's flip the ratings and the comments around. They should end up looking like this:

![dogo-movies-v1 2](https://cloud.githubusercontent.com/assets/794084/7052922/817e5a1e-dde6-11e4-9946-c0b03862cb7f.jpg)

Also, we're trying to include the images for the news! We'll keep you updated...

dogomedia commented:
We've updated the CSS in order to hide the titles and move the review count to the right hand side.
```
/* Hide the tile titles */
.zci--dogo_movies .tile__title--2,
.use-opts .zci--dogo_movies .tile__title--2 {
    display:none;
}

/* Move stars to the left */
.zci--dogo_movies .tile--b--i .stars {
    float: left;
}

/* Move review count to the right */
.zci--dogo_movies .tile--b--i .tile__source {
    float: right;
}
```
The tiles should like identical to your screenshot now.

![ddg-tiles](https://cloud.githubusercontent.com/assets/10837292/7053790/ca671ed6-ddec-11e4-828a-656b6f2a8235.png)

As a side note, we are using the default ```media``` template and it seems like the template should have a standard way of moving stars to left and count to the right simply by changing options around.  But the documentation did not seem to have a way to do this, so we simply overrode the default css for the tiles.

moollaza commented:
> But the documentation did not seem to have a way to do this, so we simply overrode the default css for the tiles.

That's perfect for now -- we're working to make the templates more easily configurable via JS.

@abeyang this is ready for another design pass. Its updated on DDH8:

moollaza commented:
@dogomedia good news, were making News images a higher priority and they'll be coming much sooner. Please add back the `image` property to the dogo_news.js `normalize()` function. When the templates are updated the images will be there :)

Here's how the updated news template will look:

![31050189451479 osvls2hfgurm3dvntzko_height640](https://cloud.githubusercontent.com/assets/873785/7097797/23490e68-dfa7-11e4-966d-e1fea96dc95f.png)

moollaza commented:
Oh and also, there's still a failing test in t/News.t -- you can run the test with `prove -Ilib t/News.t`

dogomedia commented:
@moollaza Thanks for following up with us.  We've added the image property back and fixed the test failures.

moollaza commented:
@dogomedia thanks a lot this is looking great.

I've been doing some final Q/A and I noticed there are a few relevancy issues which can be fixed in the Spice, or potentially on your end.

In the Spice, we can add a relevancy block, to compare the title of the book against the query: https://duck.co/duckduckhack/spice_displaying#relevancy

If you want to fix it at the API level that would be fine too.

Here's some queries which return irrelevant results (these all came through our autocomplete):

- https://ddh8.duckduckgo.com/?q=kids+books+to+read+online&kp=-1&ia=kidsbooks 
- https://ddh8.duckduckgo.com/?q=kids+books+for+sale&kp=-1&ia=kidsbooks
- https://ddh8.duckduckgo.com/?q=free+kids+books&kp=-1&ia=kidsbooks
- https://ddh8.duckduckgo.com/?q=kids+movies+in+theaters&kp=-1&ia=kidsmovies
- https://ddh8.duckduckgo.com/?q=kids+movies+coming+soon&kp=-1&ia=kidsmovies
- https://ddh8.duckduckgo.com/?q=kids+movies+2015&kp=-1&ia=kidsmovies
- https://ddh8.duckduckgo.com/?q=kids+newspaper+template&kp=-1&ia=kidsnews
- https://ddh8.duckduckgo.com/?q=kids+newsletter&kp=-1&ia=kidsnews 

------

Also, we've made a new template group called, 'movies' which you can use to achieve the look you have for the Dogo Books and Dogo Movies Spices. If you set `group: 'movies'` you should be able to remove the custom CSS/JS needed to hide the tile body and also the `tile: 'poster'` from the `variants` object.

> More details in this PR: https://github.com/duckduckgo/zeroclickinfo-spice/pull/1754

dogomedia commented:
@moollaza We'll fix the relevancy issues on our end at the API level in the next few days.

moollaza commented:
>We'll fix the relevancy issues on our end at the API level in the next few days.

@dogomedia awesome, I believe this is ready then :+1:

moollaza commented:
@dogomedia congrats! It's live: https://duckduckgo.com/?q=dogo+books&ia=kidsbooks

Also, https://duckduckgo.com/?q=dogo+movies and https://duckduckgo.com/?q=dogo+news

Thanks again for taking the time to contribute! We really appreciate it.

moollaza commented:
BTW the news IA should be ready -- we're going to switch from News template, to a new Media template but it should look essentially the same and will feature the pictures!
